src/main/java/com/onlinebookstore/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -17,50 +25,86 @@
         Book book = bookMapper.toModel(requestDto);
+        setCategories(book, requestDto.getCategoryIds());
         Book savedBook = bookRepository.save(book);
         return bookMapper.toDto(savedBook);
     }
 
     @Override
-    public List<BookDto> findAll(String email, Pageable pageable) {
+    public List<BookDtoWithoutCategoriesIds> findAll(Pageable pageable) {
         Page<Book> booksPage = bookRepository.findAll(pageable);
         if (booksPage.isEmpty()) {
             throw new EntityNotFoundException("Can't find any books");
         }
         List<Book> books = booksPage.getContent();
         return books.stream()
-                .map(bookMapper::toDto)
+                .map(bookMapper::toDtoWithoutCategories)
                 .toList();
     }
 
     @Override
-    public BookDto findById(Long id) {
+    public BookResponseDto findById(Long id) {
         Book book = findByIdOrThrowException(id);
         return bookMapper.toDto(book);
     }
 
     @Override
-    public BookDto update(Long id, BookRequestDto requestDto) {
+    public BookResponseDto update(Long id, BookRequestDto requestDto) {
         Book book = findByIdOrThrowException(id);
         bookMapper.updateFromDto(requestDto, book);
+        setCategories(book, requestDto.getCategoryIds());
         Book updatedBook = bookRepository.save(book);
         return bookMapper.toDto(updatedBook);
     }
 
     @Override
-    public BookDto delete(Long id) {
-        Book book = findByIdOrThrowException(id);
-        bookRepository.delete(book);
-        return bookMapper.toDto(book);
+    public void delete(Long id) {
+        findByIdOrThrowException(id);
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public List<BookDtoWithoutCategoriesIds> getBooksById(Long id, Pageable pageable) {
+        Category category = categoryService.findByIdOrThrowException(id);
+        List<BookDtoWithoutCategoriesIds> books = bookRepository
+                .findAllByCategoriesId(id, pageable).stream()
+                .map(bookMapper::toDtoWithoutCategories)
+                .collect(Collectors.toList());
+        if (books.isEmpty()) {
+            throw new EntityNotFoundException("Can't find any books for category id " + id);
+        }
+        return books;
     }
 
     private Book findByIdOrThrowException(Long id) {
         return bookRepository.findById(id)
-                .orElseThrow(() -> new EntityNotFoundException("Can't find book by id " + id));
+                .orElseThrow(() -> new EntityNotFoundException("Can't find category by id " + id));
+    }
+
+    private void setCategories(Book book, Set<Long> categoryIds) {
+        for (Long categoryId : categoryIds) {
+            if (categoryId > Long.MAX_VALUE) {
+                throw new InvalidCategoryIdException("Category ID "
+                        + categoryId
+                        + " exceeds the maximum allowed value.");
+            }

```

Comment:
i am not sure that check makes sense, I mean we are checking if a `Long` typed variable has value more than `Long.MAX_VALUE`?

Reply:
I think I misunderstand previous comment